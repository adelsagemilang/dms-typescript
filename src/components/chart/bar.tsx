import React, { Component } from 'react';
import Chart from 'react-apexcharts';
import { Options, ZoomedOptions } from './options/bar';

interface Data {
  name: string;
  data: number[];
}

interface PropTypes {
  isZoom: boolean;
  height?: number;
  data?: Data;
}

interface StateTypes {
  options: any;
  series: any;
}

const defaultSeries = [
  {
    name: 'Order',
    data: [44, 55, 57, 56, 61],
  },
];

class Bar extends Component<PropTypes, StateTypes> {
  constructor(props) {
    super(props);
    this.state = {
      series: props.series ? props.series : defaultSeries,
      options: props.isZoom ? ZoomedOptions(props) : Options(props),
    };
  }

  render() {
    const { height } = this.props;
    return (
      <div className="mixed-chart">
        <Chart
          options={this.state.options}
          series={this.state.series}
          type="bar"
          height={height || 230}
        />
      </div>
    );
  }
}

export default Bar;
