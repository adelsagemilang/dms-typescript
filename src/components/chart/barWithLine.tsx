import React, { Component } from 'react';
import Chart from 'react-apexcharts';
import { Options, ZoomedOptions } from './options/barWithLine';

interface StateTypes {
  options: any;
}

interface PropTypes {
  isZoom: boolean;
  customOptions: any;
  height?: number;
  series: any;
}

const defaultSeries = [
  {
    name: 'Total',
    type: 'column',
    data: [440, 505, 414, 671, 227],
  },
  {
    name: 'Active',
    type: 'line',
    data: [440, 505, 414, 671, 227],
  },
];

class BarWithLine extends Component<PropTypes, StateTypes> {
  constructor(props) {
    super(props);

    this.state = {
      options: props.isZoom ? ZoomedOptions(props) : Options(props),
    };
  }

  render() {
    const { height, series } = this.props;
    return (
      <div className="mixed-chart">
        <Chart
          options={this.state.options}
          series={series || defaultSeries}
          height={height || 230}
          width="100%"
          type="line"
        />
      </div>
    );
  }
}

export default BarWithLine;
