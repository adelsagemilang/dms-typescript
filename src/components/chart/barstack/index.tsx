import * as React from 'react';
import { BarChart, Bar, XAxis, YAxis, CartesianGrid, Tooltip } from 'recharts';
import { DataBar } from '../../../store/chartType';
import { defaultColors, defaultDimension } from './settings';

type ChartProps = {
  data?: DataBar[];
  colors: string[];
  width: number;
  height: number;
  toPercentage?: boolean;
};

class BarstackChart extends React.Component<ChartProps> {
  public static defaultProps = {
    colors: defaultColors,
    width: defaultDimension.width,
    height: defaultDimension.height,
  };

  render() {
    const { data } = this.props;

    return (
      <BarChart
        width={this.props.width}
        height={this.props.height}
        data={data}
        barCategoryGap={20}
      >
        <CartesianGrid vertical={false} strokeDasharray="3 3" />
        <XAxis dataKey="name" />
        <YAxis />
        <Tooltip />
        {data &&
          Object.keys(data[0]).map((entry, indexEntry) => {
            if (entry !== 'name') {
              return (
                <Bar
                  dataKey={entry}
                  stackId="a"
                  fill={this.props.colors[indexEntry - 1]}
                />
              );
            }
            return true;
          })}
      </BarChart>
    );
  }
}

export default BarstackChart;
