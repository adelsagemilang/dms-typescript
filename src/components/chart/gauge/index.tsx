import * as React from 'react';
import ReactSpeedometer from 'react-d3-speedometer';
import { DataGauge } from '../../../store/chartType';
import {
  defaultColors,
  defaultDimension,
  defaultSettings,
  defaultDataRange,
} from './settings';

type ChartProps = {
  data: DataGauge;
  colors: string[];
  width: number;
  height: number;
  ringWidth: number;
  needleColor: string;
  textColor: string;
  needleHeightRatio: number;
  dataRange: string[];
};

export default class GaugeChart extends React.Component<ChartProps> {
  public static defaultProps = {
    colors: defaultColors,
    width: defaultDimension.width,
    height: defaultDimension.height,
    ringWidth: defaultSettings.ringWidth,
    needleColor: defaultSettings.needleColor,
    textColor: defaultSettings.textColor,
    needleHeightRatio: defaultSettings.needleHeightRatio,
    dataRange: defaultDataRange,
  };

  render() {
    const { data } = this.props;
    return (
      <ReactSpeedometer
        maxValue={data.maxValue}
        value={data.value ? data.value : 0}
        width={this.props.width}
        height={this.props.height}
        needleColor={this.props.needleColor}
        ringWidth={this.props.ringWidth}
        segmentColors={this.props.colors}
        textColor={this.props.textColor}
        needleHeightRatio={this.props.needleHeightRatio}
        // customSegmentStops={data.allSegments}
        customSegmentStops={this.props.dataRange}
        valueFormat=".0%"
        currentValueText=""
      />
    );
  }
}
