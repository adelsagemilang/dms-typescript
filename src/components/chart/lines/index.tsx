import * as React from 'react';
import {
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
} from 'recharts';
import { DataLine, Margin } from '../../../store/chartType';
import { defaultColors, defaultDimension } from './settings';

type ChartProps = {
  data?: DataLine[];
  colors: string[];
  width: number;
  height: number;
  margin?: Margin;
  toPercentage?: boolean;
};

class LinesChart extends React.Component<ChartProps> {
  public static defaultProps = {
    colors: defaultColors,
    width: defaultDimension.width,
    height: defaultDimension.height,
  };

  render() {
    const { data } = this.props;

    return (
      <LineChart
        width={this.props.width}
        height={this.props.height}
        data={data}
        margin={this.props.margin}
      >
        <CartesianGrid vertical={false} strokeDasharray="3 3" />
        <XAxis dataKey="name" />
        <YAxis padding={{ top: 10 }} />
        <Tooltip />
        {data &&
          Object.keys(data[0]).map((entry, indexEntry) => {
            if (entry !== 'name') {
              return (
                <Line
                  type="linear"
                  dataKey={entry}
                  stroke={this.props.colors[indexEntry - 1]}
                  activeDot={{ r: 5 }}
                />
              );
            }
            return true;
          })}
      </LineChart>
    );
  }
}

export default LinesChart;
