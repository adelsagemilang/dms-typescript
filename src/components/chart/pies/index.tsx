import * as React from 'react';
import { PieChart, Pie, Cell } from 'recharts';
import { DataPie, Margin } from '../../../store/chartType';
import {
  defaultColors,
  defaultDimension,
  renderCustomizedLabel,
} from './settings';

type ChartProps = {
  data?: DataPie[];
  colors: string[];
  width: number;
  height: number;
  margin?: Margin;
};

class LinesChart extends React.Component<ChartProps> {
  public static defaultProps = {
    colors: defaultColors,
    width: defaultDimension.width,
    height: defaultDimension.height,
  };

  render() {
    const { data } = this.props;
    return (
      <div className="is-size-7">
        <PieChart
          width={this.props.width}
          height={this.props.height}
          margin={this.props.margin}
        >
          <Pie
            dataKey="value"
            data={data}
            labelLine={false}
            fill="#FF0000"
            label={renderCustomizedLabel}
          >
            {data &&
              data.map((entry, index) => (
                <Cell
                  key={`cell-${index}`}
                  fill={this.props.colors[index % this.props.colors.length]}
                />
              ))}
          </Pie>
        </PieChart>
      </div>
    );
  }
}

export default LinesChart;
