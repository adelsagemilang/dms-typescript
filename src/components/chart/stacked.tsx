import React, { Component } from 'react';
import Chart from 'react-apexcharts';
import { Options, ZoomedOptions } from './options/stacked';

interface StateTypes {
  options: any;
  series: any;
}

interface PropTypes {
  isZoom: boolean;
  height?: number;
  labels: string[];
}

const defaultSeries = [
  {
    name: 'Route',
    data: [44, 55, 41, 67, 22],
  },
  {
    name: 'Volume',
    data: [13, 23, 20, 8, 13],
  },
  {
    name: 'Weight',
    data: [11, 17, 15, 15, 21],
  },
];

class Stacked extends Component<PropTypes, StateTypes> {
  constructor(props) {
    super(props);

    this.state = {
      series: props.series ? props.series : defaultSeries,
      options: props.isZoom
        ? ZoomedOptions(props.labels)
        : Options(props.labels),
    };
  }

  render() {
    const { height } = this.props;
    return (
      <div className="stacked-chart">
        <Chart
          options={this.state.options}
          series={this.state.series}
          type="bar"
          height={height || 230}
        />
      </div>
    );
  }
}

export default Stacked;
