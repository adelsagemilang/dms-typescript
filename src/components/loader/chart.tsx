import React from 'react';
import ContentLoader from 'react-content-loader';

interface PropTypes {
  height: number;
  width: number;
}

const XYChart = (props: PropTypes) => {
  return (
    <ContentLoader
      viewBox={`0 0 ${props.height} ${props.width}`}
      height={props.height}
      width="100%"
      speed={1}
    >
      <rect x="20" y="175" rx="0" ry="0" width={props.width} height="1" />
      <rect x="40" y="75" rx="0" ry="0" width={props.width / 13} height="100" />
      <rect x="80" y="125" rx="0" ry="0" width={props.width / 13} height="50" />
      <rect
        x="120"
        y="105"
        rx="0"
        ry="0"
        width={props.width / 13}
        height="70"
      />
      <rect
        x="160"
        y="35"
        rx="0"
        ry="0"
        width={props.width / 13}
        height="140"
      />
      <rect
        x="200"
        y="55"
        rx="0"
        ry="0"
        width={props.width / 13}
        height="120"
      />
      <rect
        x="240"
        y="15"
        rx="0"
        ry="0"
        width={props.width / 13}
        height="160"
      />
      <rect
        x="280"
        y="135"
        rx="0"
        ry="0"
        width={props.width / 13}
        height="40"
      />
      <rect x="320" y="85" rx="0" ry="0" width={props.width / 13} height="90" />
    </ContentLoader>
  );
};

export default XYChart;
