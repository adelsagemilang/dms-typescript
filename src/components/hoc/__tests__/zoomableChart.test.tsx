import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import XYChart from 'components/loader/chart';
import { yearMonth } from 'utils/chart/commons';
import { Quantity } from 'store/quantity/types';
import Switch from 'react-switch';
import ZoomableChart from '../zoomableChart';

describe('Zoomable Chart Component Tests', () => {
  const TestWrappedComponent = () => <div className="test">Test</div>;
  const ZoomableChartComponent = ZoomableChart(TestWrappedComponent);

  const monthYear = yearMonth(2020, 3);

  it('Should renders correctly not zoom', () => {
    const mockFn = sinon.spy();
    const wrapper = mount(
      <ZoomableChartComponent
        title="Test Chart"
        toggleZoom={mockFn}
        isZoom={false}
        data={Quantity}
        chartType="stacked"
        monthYear={monthYear}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Should renders correctly while zoom', () => {
    const mockFn = sinon.spy();
    const wrapper = mount(
      <ZoomableChartComponent
        title="Test Chart"
        toggleZoom={mockFn}
        isZoom
        data={Quantity}
        chartType="stacked"
        monthYear={monthYear}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Renders Loader while fetching data', () => {
    const mockFn = sinon.spy();
    const data = {
      isLoading: true,
    };

    const wrapper = shallow(
      <ZoomableChartComponent
        title="Test Chart"
        toggleZoom={mockFn}
        isZoom={false}
        data={data}
        chartType="stacked"
        monthYear={monthYear}
      />,
    );
    expect(wrapper.find(XYChart)).toHaveLength(1);
  });

  it('Clicked Switch button', () => {
    const mockFn = sinon.spy();

    const wrapper = mount(
      <ZoomableChartComponent
        title="Test Chart"
        toggleZoom={mockFn}
        isZoom={false}
        data={Quantity}
        chartType="stacked"
        monthYear={monthYear}
      />,
    );
    wrapper
      .find(Switch)
      .find('input')
      .simulate('change');
    expect(mockFn).toHaveProperty('callCount', 1);
  });
});
