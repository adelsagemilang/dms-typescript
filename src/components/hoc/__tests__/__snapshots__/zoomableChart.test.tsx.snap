// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zoomable Chart Component Tests Should renders correctly not zoom 1`] = `
<Component
  chartType="stacked"
  data={
    Object {
      "chartData": Object {
        "data": Array [],
        "title": "",
      },
      "isLoading": false,
    }
  }
  isZoom={false}
  monthYear="2020-03"
  title="Test Chart"
  toggleZoom={[Function]}
>
  <withHelpersModifiers(Column)
    className=""
  >
    <Column
      className=""
    >
      <div
        className="column"
      >
        <withHelpersModifiers(Card)
          className="card-chart is-undefined"
        >
          <Card
            className="card-chart is-undefined"
          >
            <div
              className="card card-chart is-undefined"
            >
              <withHelpersModifiers(CardHeader)>
                <CardHeader>
                  <header
                    className="card-header"
                  >
                    <withHelpersModifiers(CardHeaderTitle)>
                      <CardHeaderTitle>
                        <p
                          className="card-header-title"
                        >
                          Test Chart
                        </p>
                      </CardHeaderTitle>
                    </withHelpersModifiers(CardHeaderTitle)>
                    <withHelpersModifiers(CardHeaderIcon)>
                      <CardHeaderIcon>
                        <a
                          className="card-header-icon"
                        >
                          <ReactSwitch
                            activeBoxShadow="0 0 2px 3px #3bf"
                            boxShadow={null}
                            checked={false}
                            checkedIcon={false}
                            className="switch"
                            disabled={false}
                            handleDiameter={17}
                            height={14}
                            id="isChecked"
                            offColor="#888"
                            offHandleColor="#fff"
                            onChange={[Function]}
                            onColor="#fff"
                            onHandleColor="#fff"
                            uncheckedIcon={false}
                            width={30}
                          >
                            <div
                              className="switch"
                              style={
                                Object {
                                  "MozTransition": "opacity 0.25s",
                                  "MozUserSelect": "none",
                                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                                  "WebkitTransition": "opacity 0.25s",
                                  "WebkitUserSelect": "none",
                                  "borderRadius": 7,
                                  "direction": "ltr",
                                  "display": "inline-block",
                                  "msUserSelect": "none",
                                  "opacity": 1,
                                  "position": "relative",
                                  "textAlign": "left",
                                  "touchAction": "none",
                                  "transition": "opacity 0.25s",
                                  "userSelect": "none",
                                }
                              }
                            >
                              <div
                                className="react-switch-bg"
                                onClick={[Function]}
                                onMouseDown={[Function]}
                                style={
                                  Object {
                                    "MozTransition": "background 0.25s",
                                    "WebkitTransition": "background 0.25s",
                                    "background": "#888888",
                                    "borderRadius": 7,
                                    "cursor": "pointer",
                                    "height": 14,
                                    "margin": 1.5,
                                    "position": "relative",
                                    "transition": "background 0.25s",
                                    "width": 30,
                                  }
                                }
                              />
                              <div
                                className="react-switch-handle"
                                onClick={[Function]}
                                onMouseDown={[Function]}
                                onTouchCancel={[Function]}
                                onTouchEnd={[Function]}
                                onTouchMove={[Function]}
                                onTouchStart={[Function]}
                                style={
                                  Object {
                                    "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                                    "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                                    "background": "#ffffff",
                                    "border": 0,
                                    "borderRadius": "50%",
                                    "boxShadow": null,
                                    "cursor": "pointer",
                                    "display": "inline-block",
                                    "height": 17,
                                    "outline": 0,
                                    "position": "absolute",
                                    "top": 0,
                                    "transform": "translateX(0px)",
                                    "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                                    "width": 17,
                                  }
                                }
                              />
                              <input
                                checked={false}
                                disabled={false}
                                id="isChecked"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                onKeyUp={[Function]}
                                role="switch"
                                style={
                                  Object {
                                    "border": 0,
                                    "clip": "rect(0 0 0 0)",
                                    "height": 1,
                                    "margin": -1,
                                    "overflow": "hidden",
                                    "padding": 0,
                                    "position": "absolute",
                                    "width": 1,
                                  }
                                }
                                type="checkbox"
                              />
                            </div>
                          </ReactSwitch>
                        </a>
                      </CardHeaderIcon>
                    </withHelpersModifiers(CardHeaderIcon)>
                  </header>
                </CardHeader>
              </withHelpersModifiers(CardHeader)>
              <withHelpersModifiers(CardImage)>
                <CardImage>
                  <div
                    className="card-image"
                  >
                    <TestWrappedComponent
                      chartType="stacked"
                      data={
                        Object {
                          "chartData": Object {
                            "data": Array [],
                            "title": "",
                          },
                          "isLoading": false,
                        }
                      }
                      isZoom={false}
                      labels={
                        Array [
                          "Nov",
                          "Dec",
                          "Jan",
                          "Feb",
                          "Mar",
                        ]
                      }
                      monthYear="2020-03"
                      series={
                        Array [
                          Object {
                            "data": Array [
                              0,
                              0,
                              0,
                              0,
                              0,
                            ],
                            "name": "Route",
                          },
                          Object {
                            "data": Array [
                              0,
                              0,
                              0,
                              0,
                              0,
                            ],
                            "name": "Volume",
                          },
                          Object {
                            "data": Array [
                              0,
                              0,
                              0,
                              0,
                              0,
                            ],
                            "name": "Weight",
                          },
                        ]
                      }
                      title="Test Chart"
                      toggleZoom={[Function]}
                    >
                      <div
                        className="test"
                      >
                        Test
                      </div>
                    </TestWrappedComponent>
                  </div>
                </CardImage>
              </withHelpersModifiers(CardImage)>
            </div>
          </Card>
        </withHelpersModifiers(Card)>
      </div>
    </Column>
  </withHelpersModifiers(Column)>
</Component>
`;

exports[`Zoomable Chart Component Tests Should renders correctly while zoom 1`] = `
<Component
  chartType="stacked"
  data={
    Object {
      "chartData": Object {
        "data": Array [],
        "title": "",
      },
      "isLoading": false,
    }
  }
  isZoom={true}
  monthYear="2020-03"
  title="Test Chart"
  toggleZoom={[Function]}
>
  <withHelpersModifiers(Column)
    className="is-zoom"
  >
    <Column
      className="is-zoom"
    >
      <div
        className="column is-zoom"
      >
        <withHelpersModifiers(Card)
          className="card-chart is-undefined"
        >
          <Card
            className="card-chart is-undefined"
          >
            <div
              className="card card-chart is-undefined"
            >
              <withHelpersModifiers(CardHeader)>
                <CardHeader>
                  <header
                    className="card-header"
                  >
                    <withHelpersModifiers(CardHeaderTitle)>
                      <CardHeaderTitle>
                        <p
                          className="card-header-title"
                        >
                          Test Chart
                        </p>
                      </CardHeaderTitle>
                    </withHelpersModifiers(CardHeaderTitle)>
                    <withHelpersModifiers(CardHeaderIcon)>
                      <CardHeaderIcon>
                        <a
                          className="card-header-icon"
                        >
                          <ReactSwitch
                            activeBoxShadow="0 0 2px 3px #3bf"
                            boxShadow={null}
                            checked={true}
                            checkedIcon={false}
                            className="switch"
                            disabled={false}
                            handleDiameter={17}
                            height={14}
                            id="isChecked"
                            offColor="#888"
                            offHandleColor="#fff"
                            onChange={[Function]}
                            onColor="#fff"
                            onHandleColor="#fff"
                            uncheckedIcon={false}
                            width={30}
                          >
                            <div
                              className="switch"
                              style={
                                Object {
                                  "MozTransition": "opacity 0.25s",
                                  "MozUserSelect": "none",
                                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                                  "WebkitTransition": "opacity 0.25s",
                                  "WebkitUserSelect": "none",
                                  "borderRadius": 7,
                                  "direction": "ltr",
                                  "display": "inline-block",
                                  "msUserSelect": "none",
                                  "opacity": 1,
                                  "position": "relative",
                                  "textAlign": "left",
                                  "touchAction": "none",
                                  "transition": "opacity 0.25s",
                                  "userSelect": "none",
                                }
                              }
                            >
                              <div
                                className="react-switch-bg"
                                onClick={[Function]}
                                onMouseDown={[Function]}
                                style={
                                  Object {
                                    "MozTransition": "background 0.25s",
                                    "WebkitTransition": "background 0.25s",
                                    "background": "#ffffff",
                                    "borderRadius": 7,
                                    "cursor": "pointer",
                                    "height": 14,
                                    "margin": 1.5,
                                    "position": "relative",
                                    "transition": "background 0.25s",
                                    "width": 30,
                                  }
                                }
                              />
                              <div
                                className="react-switch-handle"
                                onClick={[Function]}
                                onMouseDown={[Function]}
                                onTouchCancel={[Function]}
                                onTouchEnd={[Function]}
                                onTouchMove={[Function]}
                                onTouchStart={[Function]}
                                style={
                                  Object {
                                    "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                                    "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                                    "background": "#ffffff",
                                    "border": 0,
                                    "borderRadius": "50%",
                                    "boxShadow": null,
                                    "cursor": "pointer",
                                    "display": "inline-block",
                                    "height": 17,
                                    "outline": 0,
                                    "position": "absolute",
                                    "top": 0,
                                    "transform": "translateX(16px)",
                                    "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                                    "width": 17,
                                  }
                                }
                              />
                              <input
                                checked={true}
                                disabled={false}
                                id="isChecked"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                onKeyUp={[Function]}
                                role="switch"
                                style={
                                  Object {
                                    "border": 0,
                                    "clip": "rect(0 0 0 0)",
                                    "height": 1,
                                    "margin": -1,
                                    "overflow": "hidden",
                                    "padding": 0,
                                    "position": "absolute",
                                    "width": 1,
                                  }
                                }
                                type="checkbox"
                              />
                            </div>
                          </ReactSwitch>
                        </a>
                      </CardHeaderIcon>
                    </withHelpersModifiers(CardHeaderIcon)>
                  </header>
                </CardHeader>
              </withHelpersModifiers(CardHeader)>
              <withHelpersModifiers(CardImage)>
                <CardImage>
                  <div
                    className="card-image"
                  >
                    <TestWrappedComponent
                      chartType="stacked"
                      data={
                        Object {
                          "chartData": Object {
                            "data": Array [],
                            "title": "",
                          },
                          "isLoading": false,
                        }
                      }
                      isZoom={true}
                      labels={
                        Array [
                          "Nov",
                          "Dec",
                          "Jan",
                          "Feb",
                          "Mar",
                        ]
                      }
                      monthYear="2020-03"
                      series={
                        Array [
                          Object {
                            "data": Array [
                              0,
                              0,
                              0,
                              0,
                              0,
                            ],
                            "name": "Route",
                          },
                          Object {
                            "data": Array [
                              0,
                              0,
                              0,
                              0,
                              0,
                            ],
                            "name": "Volume",
                          },
                          Object {
                            "data": Array [
                              0,
                              0,
                              0,
                              0,
                              0,
                            ],
                            "name": "Weight",
                          },
                        ]
                      }
                      title="Test Chart"
                      toggleZoom={[Function]}
                    >
                      <div
                        className="test"
                      >
                        Test
                      </div>
                    </TestWrappedComponent>
                  </div>
                </CardImage>
              </withHelpersModifiers(CardImage)>
            </div>
          </Card>
        </withHelpersModifiers(Card)>
      </div>
    </Column>
  </withHelpersModifiers(Column)>
</Component>
`;
